experimental/framework/last-edited/src/setup.ts

```diff
@@ -28,14 +28,16 @@ function getLastEditDetailsFromMessage(
 	message: ISequencedDocumentMessage,
 	quorum: IQuorumClients,
 ): ILastEditDetails | undefined {
-	const sequencedClient = quorum.getMember(message.clientId);
-	const user = sequencedClient?.client.user;
-	if (user !== undefined) {
-		const lastEditDetails: ILastEditDetails = {
-			user,
-			timestamp: message.timestamp,
-		};
-		return lastEditDetails;
+	if (message.clientId !== null) {
```

Comment:
This seems fine for this change, but also makes me think that it wants the server messages filtered out earlier in the pipeline.

Reply:
maybe. this get's scheduled on the runtime op event, which i think includes summary ops, which will have null client id. the op check saved us in that case, but its not like this code never sees null client id, it just didn't have to deal with it before the type was fixed.

Reply:
the runtime could probably support a discriminated union of ops types and what their shape is, but that is beyond the scope of this change